src/mpyl/steps/deploy/kubernetes.py

```diff
@@ -85,7 +85,7 @@ def execute(self, step_input: Input) -> Output:
                     self._logger,
                     chart,
                     step_input,
-                    Target.PRODUCTION,
+                    Target.PULL_REQUEST_BASE,
```

Comment:
Wow! We were deploying to prod twice? :-)
Maybe it would be cleaner to do this in the runner instead of the step?

Reply:
Yeah maybe we can remove this logic from mpyl.. It's a bit implicit, and with the proper caching mechanism in place it shouldn't be needed either

Reply:
Agreed. It's too implicit 👍 
Why would a proper caching mechanism make this logic redundant?

Reply:
I thought we're doing this to ensure the same image gets deployed to test which is in prod. This already was the case when the work is scheduled on the same agent. With the artifact repo, it will always be the case right?

Reply:
Ah, I understand what you mean by caching now. I'm not sure about that. Because although the images used in test should be identical to production, the config will not be.

Reply:
wait.. currently if you're scheduled on the same agent and you're deploying a tag it's not rebuilding the image